main.py

```diff
@@ -31,28 +48,94 @@ def save_entries(entries):
         json.dump([asdict(entry) for entry in entries], file, indent=2)
 
 
+# type for the command line interface
+app = typer.Typer()
+
+
 # ➕ Add a new entry
-def add_entry(title, content):
+@app.command()
+def add(
+    title: str = typer.Option(..., prompt="Title", help="Title of the journal entry"),
+    content: str = typer.Option(
+        ..., prompt="Content", help="Content of the journal entry"
+    ),
+):
+    """➕ Add a new journal entry."""
+    typer.echo(f"Added entry: {title} - {content} successfully.")
     entries = load_entries()
     new_entry = JournalEntry(title=title, content=content)
     entries.append(new_entry)
     save_entries(entries)
 
 
-# 📃 List all entries
-def list_entries(entries):
-    for entry in entries:
-        print(f"\n📅 {entry.date}")
-        print(f"📓 {entry.title}")
-        print(entry.content)
-        print("-" * 30)
+# 📘 List journal entries
+@app.command()
+def list():
+    """📘 List all journal entries."""
+    entries = load_entries()
+    if not entries:
+        typer.echo("📭 No entries found.")
+    else:
+        typer.echo("📘 Your Journal:")
+        n = 0
+        # Iterate through entries and print them
+        for entry in entries:
+            n = n + 1
+            print(f"\n{n} 📅 {entry.date}")
+            print(f"📓 {entry.title}")
+            print(entry.content)
+            print("-" * 30)
+
+
+# 📊 Count total entries
+@app.command()
+def count():
+    """📊 Count total journal entries."""
+    entries = load_entries()
+    count = len(entries)
+    typer.echo(f"\n 📊 Total entries 👁  in the Journal🟰 {count}")
+
+
+# Exit the application
+@app.command()
+def exit():
+    """🙌🏼 Exit the Journal application."""
+    typer.echo("🙌🏼 Thank you for using the Journal! Goodbye!")
+    typer.Exit(code=0)
+
+
+# prompt user for CLI commands
+def interactive_menu():
+    typer.echo(
+        "Welcome to your Journal! 📖"
+        "\nChoose an option:"
+        "\n1. ➕ Add a new entry: add"
+        "\n2. 📘 List all entries: list"
+        "\n3. 📊 Count total entries: count"
+        "\n4. 🙌🏼 Exit"
+        "\n5. 🛟 Help"
+    )
+    choice = typer.prompt("Enter your choice (1-4)", type=int)
+    if choice == 1:
+        add()
+    elif choice == 2:
+        list()
+    elif choice == 3:
+        count()
+    elif choice == 4:
+        exit()
+    elif choice == 5:
```

Comment:
there is no validation for range right? so in that case add an `else` for catch-all